avocado/utils/lv_utils.py

```diff
@@ -0,0 +1,517 @@
+        result = process.run("vgcreate %s %s" %
+                             (vg_name, loop_device), shell=True)
+    except process.CmdError, ex:
+        LOGGER.error(ex)
+        vg_ramdisk_cleanup(ramdisk_filename, vg_ramdisk_dir,
+                           vg_name, loop_device)
+        raise ex
+
+    LOGGER.debug(result.stdout.rstrip())
+
+
+def vg_ramdisk_cleanup(ramdisk_filename=None, vg_ramdisk_dir=None,
+                       vg_name=None, loop_device=None):
+    """
+    Inline cleanup function in case of test error.
+    """
+    if vg_name is not None:
+        loop_device = re.search(r"([/w]+) %s lvm2" % vg_name,
+                                process.run("pvs").stdout)
+        if loop_device is not None:
+            loop_device = loop_device.group(1)
+        result = process.run("vgremove %s" %
+                             vg_name, ignore_status=True, shell=True)
+        if result.exit_status == 0:
+            LOGGER.debug(result.stdout.rstrip())
+        else:
+            LOGGER.debug("%s -> %s", result.command, result.stderr)
+
+    if loop_device is not None:
+        result = process.run("pvremove %s" % loop_device,
+                             ignore_status=True, shell=True)
+        if result.exit_status == 0:
+            LOGGER.debug(result.stdout.rstrip())
+        else:
+            LOGGER.debug("%s -> %s", result.command, result.stderr)
+
+        if loop_device in process.run("losetup --all").stdout:
+            ramdisk_filename = re.search(r"%s: [d+]:d+ (([/w]+))" %
+                                         loop_device,
+                                         process.run("losetup --all").stdout)
+            if ramdisk_filename is not None:
+                ramdisk_filename = ramdisk_filename.group(1)
+
+            for _ in range(10):
+                time.sleep(0.1)
+                result = process.run("losetup -d %s" % loop_device,
+                                     ignore_status=True, shell=True)
+                if "resource busy" not in result.stderr:
+                    if result.exit_status == 0:
+                        LOGGER.debug("Loop device %s deleted", loop_device)
+                    else:
+                        LOGGER.debug(
+                            "%s -> %s", result.command, result.stderr)
+                    break
+
+    if ramdisk_filename is not None:
+        if os.path.exists(ramdisk_filename):
+            os.unlink(ramdisk_filename)
+            LOGGER.debug("Ramdisk filename %s deleted", ramdisk_filename)
+            vg_ramdisk_dir = os.path.dirname(ramdisk_filename)
+
+    if vg_ramdisk_dir is not None:
+        process.run("umount %s" % vg_ramdisk_dir,
+                    ignore_status=True, shell=True)
+        if result.exit_status == 0:
+            LOGGER.debug("Successfully unmounted tmpfs from %s",
+                         vg_ramdisk_dir)
+        else:
+            LOGGER.debug("%s -> %s", result.command, result.stderr)
```

Comment:
Similar to other complains, those information are already there, so explanation of what happened would be preferred.